neuralpredictors/layers/cores/base.py

```diff
@@ -51,3 +52,42 @@ def __repr__(self):
         for attr in filter(lambda x: "gamma" in x or "skip" in x, dir(self)):
             ret.append(f"{attr} = {getattr(self, attr)}")
         return s + "|".join(ret) + "]\n"
+
+
+class ConvCore(Core):
+    def __init__(self) -> None:
+        """
+        Derived classes need to define "batch_norm", "hidden_channels", "independent_bn_bias", "momentum" attributes.
+        """
+        super().__init__()
+        self.set_batchnorm_type()
+
+    @abstractmethod
+    def set_batchnorm_type(self):
+        """
+        Set batchnorm_layer_cls, bias_layer_cls, scale_layer_cls class attributes
+        """
+        self.batchnorm_layer_cls = None
+        self.bias_layer_cls = None
+        self.scale_layer_cls = None
+
+    def add_bn_layer(self, layer: OrderedDict, layer_idx: int):
+        for attr in ["batch_norm", "hidden_channels", "independent_bn_bias", "momentum"]:
+            if not hasattr(self, attr):
+                raise NotImplementedError(f"Subclasses must have a `{attr}` attribute.")
```

Comment:
There is `batch_norm_scale` missing here

could we also add the checks like `assert(astype(hidden_channels, list))` here as well
this relates to 
- hidden_channels
- batch_norm
- batch_norm_scale